### DISCUSSION TITLE
User sequence generators in helpers

### DISCUSSION THREAD

**Discussion by if-mar** [vitaly-t/pg-promise#879] [Q&A] [answered]
I would like to use the pgp insert helper to generate an insert query. However, I neet to generate an id using a sequence generator. How would I tell this to pgp helper?

The "handmade" query would look like 
```sql
insert into border(id, timestamp, sensor_device_id, value) VALUES(nextVal('sequence_border'), '2023-04-17T08:19:00.000Z', 'ABCDE', 0)
```

How can i achive that with the insert helper function?
```typescript
const borderCrossingColumns = new pgp.helpers.ColumnSet(['id', 'timestamp', 'sensor_device_id', 'value'],
    {table: 'border_crossing'});

const query = pgp.helpers.insert(bordersArray, borderCrossingColumns)
```

**Comment by vitaly-t** [comment, marked as answer]
Your `id` column should be:

```js
{
    name: 'id',
    mod: ':raw',
    init: () => pgp.as.format('nextVal($1)', ['sequence_border'])
}
```

Or even simpler, if the column text is static:

```js
{
    name: 'id',
    mod: ':raw',
    init: () => 'nextVal(\'sequence_border\')'
}
```

See [Column](https://vitaly-t.github.io/pg-promise/helpers.Column.html) class.

**Reply by if-mar** [reply]
Thank you very much for your quick reply! It now works as expected.
